docs/reference/enrich-processor/normalize-for-stream.md

```diff
@@ -151,3 +151,105 @@ will be normalized into the following form:
   "trace_id": "abcdef1234567890abcdef1234567890"
 }
 ```
+## Structured `message` field
+
+If the `message` field in the ingested document is structured as a JSON, the
+processor will determine whether it is in ECS format or not, based on the
+existence or absence of the `@timestamp` field. If the `@timestamp` field is
+present, the `message` field will be considered to be in ECS format, and its
+contents will be merged into the root of the document and then normalized as
+described above. The `@timestamp` from the `message` field will override the
+root `@timestamp` field in the resulting document.
+If the `@timestamp` field is absent, the `message` field will be moved to
+the `body.structured` field as is, without any further normalization.
+
+For example, if the `message` field is an ECS-JSON, as follows:
+
+```json
+{
+  "@timestamp": "2023-10-01T12:00:00Z",
+  "message": {
+    "@timestamp": "2023-10-02T12:00:00Z",
+    "log.level": "INFO",
+    "service.name": "my-service",
+    "message": "The actual log message",
```

Comment:
This isn't the main case this is about, right? It's about message being a string which can be parsed as json. I guess we should make an example like
```
```json
{
  "@timestamp": "2023-10-01T12:00:00Z",
  "message": "{\"@timestamp\":\"2023-10-02T12:00:00Z\",\"log.level\":\"INFO\",\"service.name\":\"my-service\",\"message\":\"The actual log message\",\"http\":{\"method\":\"GET\",\"url\":{\"path\":\"/api/v1/resource\"}}}"
```

it's a little messy, but that's kind of the point.

Reply:
Yes, you are right! I fixed that in the javadoc of the tests and forgot to fix it here. Thanks for pointing this out!